opam install --build-test fails

Hello, here's the output I got from opam:

```
> opam install --build-test mirage-clock-lwt
The following actions will be performed:
  ∗  install conf-m4          1               [required by ocamlfind]
  ∗  install result           1.2             [required by lwt]
  ∗  install ocamlbuild       0.11.0          [required by mirage-clock-lwt]
  ∗  install ocamlfind        1.7.1           [required by mirage-clock-lwt]
  ∗  install uchar            0.0.1           [required by fmt]
  ∗  install topkg            0.9.0           [required by mirage-clock-lwt]
  ∗  install ppx_tools        5.0             [required by lwt]
  ∗  install fmt              0.8.2           [required by mirage-device]
  ∗  install lwt              2.7.0           [required by mirage-clock-lwt]
  ∗  install mirage-device    1.0.0           [required by mirage-clock]
  ∗  install mirage-clock     1.2.0           [required by mirage-clock-lwt]
  ∗  install mirage-clock-lwt 1.2.0 
===== ∗  12 =====
Do you want to continue ? [Y/n] y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[fmt] Archive in cache
[lwt] Archive in cache
[mirage-clock] Archive in cache
[mirage-clock-lwt] Archive in cache
[mirage-device] Archive in cache
[ocamlbuild] Archive in cache
[ocamlfind] Archive in cache
[ppx_tools] Archive in cache
[result] Archive in cache
[topkg] Archive in cache
[uchar] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
∗  installed conf-m4.1
∗  installed result.1.2
∗  installed ocamlfind.1.7.1
∗  installed ppx_tools.5.0
∗  installed ocamlbuild.0.11.0
∗  installed uchar.0.0.1
∗  installed topkg.0.9.0
∗  installed fmt.0.8.2
∗  installed mirage-device.1.0.0
∗  installed mirage-clock.1.2.0
∗  installed lwt.2.7.0
[ERROR] The compilation of mirage-clock-lwt failed at "ocaml pkg/pkg.ml test".

#=== ERROR while installing mirage-clock-lwt.1.2.0 ============================#
# opam-version 1.2.2
# os           linux
# command      ocaml pkg/pkg.ml test
# path         /home/bertrand/.opam/mirage-issue/build/mirage-clock-lwt.1.2.0
# compiler     4.04.0
# exit-code    1
# env-file     /home/bertrand/.opam/mirage-issue/build/mirage-clock-lwt.1.2.0/mirage-clock-lwt-7288-a08ad8.env
# stdout-file  /home/bertrand/.opam/mirage-issue/build/mirage-clock-lwt.1.2.0/mirage-clock-lwt-7288-a08ad8.out
# stderr-file  /home/bertrand/.opam/mirage-issue/build/mirage-clock-lwt.1.2.0/mirage-clock-lwt-7288-a08ad8.err
### stderr ###
# pkg.ml: [ERROR] _build/_topkg-mirage-clock.tests: no such file. Did you forget to build the package ?
```

--- a/mirage-clock-lwt.opam
+++ b/mirage-clock-lwt.opam
@@ -18,5 +18,5 @@ depends: [
 build: [ "ocaml" "pkg/pkg.ml" "build" "--pkg-name" name "--pinned" "%{pinned}%" ]
 build-test: [
   [ "ocaml" "pkg/pkg.ml" "build" "--pkg-name" name "--pinned" "%{pinned}%" "--tests" "true" ]
-  [ "ocaml" "pkg/pkg.ml" "test" ]
+  [ "ocaml" "pkg/pkg.ml" "test" "--pkg-name" name ]
 ]

--- a/mirage-clock-unix.opam
+++ b/mirage-clock-unix.opam
@@ -22,5 +22,5 @@ build: [
 ]
 build-test: [
   [ "ocaml" "pkg/pkg.ml" "build" "-n" name "--pinned" "%{pinned}%" "--tests" "true" ]
-  [ "ocaml" "pkg/pkg.ml" "test" ]
+  [ "ocaml" "pkg/pkg.ml" "test" "-n" name ]
 ]

